launch_testing/launch_testing/__init__.py

```diff
@@ -35,6 +35,7 @@ def __init__(
         self.name = name
         self.launch_descriptor = launch_descriptor
         self.expected_lines = expected_lines
+        self.expected_output = b'\n'.join(self.expected_lines)
```

Comment:
But if you anchor your regex with a leading `^` it won't work anymore. And without that you might match other lines.

I am just saying that the current code works only fine in a very narrow set of use cases.